app/src/ui/changes/undo-commit.tsx

```diff
@@ -24,6 +24,24 @@ interface IUndoCommitProps {
 
 /** The Undo Commit component. */
 export class UndoCommit extends React.Component<IUndoCommitProps, {}> {
+  private button: HTMLButtonElement | null = null
+
+  public componentDidMount(): void {
+    const diff = this.props.commit.author.date.getTime() - Date.now()
+    const duration = Math.abs(diff)
+    if (duration < 1000 && this.props.isCommitting === false) {
+      this.focusButton()
+    }
```

Comment:
What's the reasoning behind this? Because I'm think this could not work if a commit + "refreshing the repo" takes longer than 1s (which I think happens more often on Windows) 🤔

Reply:
Yeah. This part I wasn't too happy about, but was having trouble finding a better way.